src/views/scroll-editor/manuscript-toolbar.vue

```diff
@@ -500,7 +528,63 @@ export default class ManuscriptToolbar extends Vue {
         this.params.mode = mode;
     }
 
+    private getArtefactHeight(artefact?: Artefact) {
+        if (artefact) {
+            const pointsAttribute = artefact.mask.svg;
+            const points = pointsAttribute.split('L');
+            let maxY = -Infinity;
+            let minY = Infinity;
+            for (const point of points) {
+                const [x, y] = point.split(' ').map(n => parseFloat(n));
+                maxY = Math.max(maxY, y);
+                minY = Math.min(minY, y);
+            }
+            const maxHeight = maxY - minY;
+            return Math.round((maxHeight / this.ppi * 2.54) * 100) / 100;
+        }
+    }
+    private getArtefactWidth(artifact?: Artefact) {
+        if (artifact) {
+            const pointsAttribute = artifact.mask.svg;
+            const points = pointsAttribute.split('L');
+            let maxX = -Infinity;
+            let minX = Infinity;
+            for (let point of points) {
+                if (point[0] === 'M') {
+                    point = point.substring(1);
+                }
+                const [x, y] = point.split(' ').map(n => parseFloat(n));
+                maxX = Math.max(maxX, x);
+                minX = Math.min(minX, x);
+            }
+            const maxWidth = maxX - minX;
+            return Math.round((maxWidth / this.ppi * 2.54) * 100) / 100;
+        }
+    }
+    private convertToMM(number: number) {
+        return Math.round((number / this.ppi * 2.54) * 100) / 100;
+    }
+    private getGroupSize() {
```

Comment:
I think it's better to calculate the group size when the group changes, and store the value in the object. Then bind to that value. Otherwise this function will be called too many times. Same for the artefact height and width.